### ISSUE TITLE
A simple JSoup processing exception killed the generator application

### ISSUE THREAD

**Issue by laxika** [bottomless-archive-project/library-of-alexandria#254]
```
2021-01-14 12:35:22.262 ERROR 19096 --- [ing-scheduler-1] .c.s.l.CommonCrawlDocumentLocationSource : Failed to download WARC location https://commoncrawl.s3.amazonaws.com/crawl-data/CC-MAIN-2020-50/segments/1606141163411.0/warc/CC-MAIN-20201123153826-20201123183826-00001.warc.gz!

java.lang.NullPointerException: Cannot invoke "org.jsoup.nodes.Attributes.get(String)" because "startTag.attributes" is null
        at org.jsoup.parser.HtmlTreeBuilderState$9.process(HtmlTreeBuilderState.java:934) ~[jsoup-1.13.1.jar!/:na]
        at org.jsoup.parser.HtmlTreeBuilder.process(HtmlTreeBuilder.java:136) ~[jsoup-1.13.1.jar!/:na]
        at org.jsoup.parser.HtmlTreeBuilderState$10.process(HtmlTreeBuilderState.java:1019) ~[jsoup-1.13.1.jar!/:na]
        at org.jsoup.parser.HtmlTreeBuilder.process(HtmlTreeBuilder.java:136) ~[jsoup-1.13.1.jar!/:na]
        at org.jsoup.parser.TreeBuilder.runParser(TreeBuilder.java:66) ~[jsoup-1.13.1.jar!/:na]
        at org.jsoup.parser.TreeBuilder.parse(TreeBuilder.java:47) ~[jsoup-1.13.1.jar!/:na]
        at org.jsoup.parser.Parser.parse(Parser.java:107) ~[jsoup-1.13.1.jar!/:na]
        at org.jsoup.Jsoup.parse(Jsoup.java:31) ~[jsoup-1.13.1.jar!/:na]
        at com.github.loa.source.commoncrawl.service.WarcRecordParser.lambda$parseLinksFromRecord$0(WarcRecordParser.java:20) ~[loa-source-commoncrawl-service-1.1.0-release-candidate.2.jar!/:na]
        at reactor.core.publisher.MonoSupplier.call(MonoSupplier.java:85) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at reactor.core.publisher.MonoFlattenIterable.subscribeOrReturn(MonoFlattenIterable.java:68) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at reactor.core.publisher.Flux.subscribe(Flux.java:8133) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at reactor.core.publisher.FluxFlatMap$FlatMapMain.onNext(FluxFlatMap.java:425) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at reactor.core.publisher.FluxSubscribeOn$SubscribeOnSubscriber.onNext(FluxSubscribeOn.java:150) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext(FluxMapFuseable.java:127) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at reactor.core.publisher.FluxIterable$IterableSubscription.slowPath(FluxIterable.java:270) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at reactor.core.publisher.FluxIterable$IterableSubscription.request(FluxIterable.java:228) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.request(FluxMapFuseable.java:169) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at reactor.core.publisher.FluxSubscribeOn$SubscribeOnSubscriber.requestUpstream(FluxSubscribeOn.java:130) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at reactor.core.publisher.FluxSubscribeOn$SubscribeOnSubscriber.onSubscribe(FluxSubscribeOn.java:123) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onSubscribe(FluxMapFuseable.java:96) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at reactor.core.publisher.FluxIterable.subscribe(FluxIterable.java:164) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at reactor.core.publisher.FluxStream.subscribe(FluxStream.java:71) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at reactor.core.publisher.InternalFluxOperator.subscribe(InternalFluxOperator.java:62) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at reactor.core.publisher.FluxSubscribeOn$SubscribeOnSubscriber.run(FluxSubscribeOn.java:193) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37) ~[reactor-core-3.4.1.jar!/:3.4.1]
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[na:na]
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[na:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[na:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[na:na]
        at java.base/java.lang.Thread.run(Thread.java:832) ~[na:na]
```

**Comment by laxika** [comment]
Blocked by https://github.com/jhy/jsoup/issues/1489.

**Comment by laxika** [comment]
The bug is fixed in JSoup 1.14.1. It is not released yet. As soon as it is released, we will need to upgrade to it.
